javascript/net/grpc/web/grpc_generator.cc

```diff
@@ -1356,43 +1356,106 @@ void PrintServerStreamingCall(Printer* printer, std::map<string, string> vars) {
   printer->Print("};\n\n\n");
 }
 
-class GrpcCodeGenerator : public CodeGenerator {
+class GeneratorOptions {
  public:
-  GrpcCodeGenerator() {}
-  ~GrpcCodeGenerator() override {}
+  GeneratorOptions();
+
+  bool ParseFromOptions(const string& parameter, string* error);
+  bool ParseFromOptions(const std::vector<std::pair<string, string>>& options,
+                        string* error);
+
+  // Returns the name of the output file for |proto_file|.
+  string OutputFile(const string& proto_file) const;
+
+  string mode() const { return mode_; }
+  ImportStyle import_style() const { return import_style_; }
+  bool generate_dts() const { return generate_dts_; }
+  bool multiple_files() const { return multiple_files_; }
+
+ private:
+  string file_name_;
+  string mode_;
+  ImportStyle import_style_;
+  bool generate_dts_;
+  bool multiple_files_;
+};
 
-  bool Generate(const FileDescriptor* file, const string& parameter,
-                GeneratorContext* context, string* error) const override {
-    std::vector<std::pair<string, string> > options;
-    ParseGeneratorParameter(parameter, &options);
-
-    string file_name;
-    string mode;
-    string import_style_str;
-    ImportStyle import_style;
-    bool generate_dts = false;
-    bool gen_multiple_files = false;
-
-    for (size_t i = 0; i < options.size(); ++i) {
-      if (options[i].first == "out") {
-        file_name = options[i].second;
-      } else if (options[i].first == "mode") {
-        mode = options[i].second;
-      } else if (options[i].first == "import_style") {
-        import_style_str = options[i].second;
-      } else if (options[i].first == "multiple_files") {
-        gen_multiple_files = options[i].second == "true";
+GeneratorOptions::GeneratorOptions()
+    : file_name_(""),
+      mode_(""),
+      import_style_(ImportStyle::CLOSURE),
+      generate_dts_(false),
+      multiple_files_(false){};
```

Comment:
Nit: our internal c++ linter doesn't like this `;` after a `}`. Please remove it. Thanks!